I want to use the Browse Everything gem

banurekha opened the pull request:
Pull request for browse everything gem implementation. This covers all sub task that are closed within the store https://jira.duraspace.org/browse/HYDRASIR-218

Xiaoming committed:
install and configue browse-everything gem [Skip ci]

Making browse-everything work with curate

[skip ci]

Adding Spec for browse everything

[skip ci]

Xiaoming committed:
View change to trigger browse-everything-not done yet

[skip ci]

Add more view code to select file

[skip ci]

Add browse-everything link to new etds and images page

[skip ci]

Adds Browse_Everything button to form

Moves browse_everything javascript to assets folder

Add missing line

[skip ci]

Removes javascript from view

Changes js file to select_everything.js

Integrates banurekha javascript work

Moving browse everything to a partial and integrating with other works

Adding cloud resource attribute to all work

adding cloud resource attachment to actor

refactor view and script to get urls to download from and store urls to hidden field

Conflicts:
	app/views/curation_concern/images/_form_files_and_links.html.erb

Xiaoming committed:
fix error for attaching multiple files

Edit page should not show Browse button

Add browse everything partial to Attach a File page

adding cloud resources module to generic file as well

removing cloud_resource module since this attribute wont be used anymore

not using js to parse the result from browse_everything

adding cloud resource model with attr that match  browse_everything gem result

reworking on actor to ingest cloud resource

hooking up cloud_resouces params and processing it.

fixing issue related to google drive

adding config for hydracurate account in cloud

code refactor to merge cloud resource attributes to be used by actor

removing duplicate extension on filename

ability to upload cloud resource for generic file

jeremyf committed:
Replacing spec:show_matchers with yenta

https://github.com/jeremyf/rspec-yenta

scherztc committed:
Adds spec test for cloud_resource model

adding spec for cloud resource

adding spec for base controller to respond to cloud resource

moving common helper methods to base controller

more cloud resource spec

defaulting file system to spec/support folder to run features

feature to add cloud resource

Adds generic_file_actor_spec

adding feature to attach a cloud resource to existing work

removing debug statements

removing debug statements

Updates generic_file_actor_spec to test for files attached from cloud.

Check if only one cloud resource is chosen and  remove alert statement from js

Add browse-everything link to new etds and images page

[skip ci]

Unit tests for download_content_from_host. A lot of stubbing and mocking was needed

Conflicts:
	app/controllers/curation_concern/generic_works_controller.rb
	dump.rdb

nomadicoder committed:
Change tab based on active type field

HYDRASIR-297 #closes

paclough committed:
Add button to clear all limits from search

Add a 'Clear limits' button to the catalog/index page within the
existing filter UI. The button is actually a link styled as a button,
since a real button sends a POST. The link redirects to the root path.
Add specs to feature test to verify functionality.

HYDRASIR-271 #close

nomadicoder committed:
Filter on non-person/collection facets

HYDRASIR-272
Uses explicit query hash for add_facet_field on
non-person/non-collection facets. This query should probably be creted
by enumerating through all of the types of works.
Also, does not filter out persons and collections if you hit the "More
Types of Works>>" link when there are more than five facets available.

hortongn committed:
Repository managers can edit users and profiles

Override the update method in
app/controllers/registrations_controller.rb so
managers can save profile changes without
entering a password.

Create app/controllers/users_controller.rb to
allow Devise to edit other users' profiles.

Add update_without_password and
update_user_without_user methods to
app/models/account.rb to allow proper updating
of user accounts.

Remove code from app/models/concerns/curate/ability.rb
that blocks person and profile update/edit/destroy
abilities from managers.

Tweak the manager_usernames method in
app/models/curate/user_behavior/base.rb so the
app does not crash if the yaml file is missing.

Edit several views to display proper buttons
and labels depeding on if they are a manager or not.
- app/views/curate/people/show.html.erb
- app/views/registrations/_form_account_deactivation.html.erb
- app/views/registrations/_form_attributes.html.erb
- app/views/registrations/_form_password_management.html.erb
- app/views/registrations/edit.html.erb

Add new /users/:id/edit route to
lib/curate/rails/routes.rb to allow managers to
edit another user's profile page.

Edit existing spec tests to adjust for changes
to buttons and other labels
- spec/features/person_profile_spec.rb
- spec/features/user_profile_workflow_spec.rb

HYDRASIR-302 #close
HYDRASIR-303 #close
HYDRASIR-313 #close
HYDRASIR-314 #close

rbalekai committed:
This part was left out for Group Collaboration

Ref: HYDRASIR-184

rbalekai committed:
Correct minor glitch in making Group title unique

spr7b committed:
HYDRASIR-36 Display Creative Commons licenses or similar license terms that submitter has selected

 1. Added a line to the _attributes.html.erb partials to display the rights (i.e. content license).
 2. Modified the curate_helper.rb curation_concern_attribute_to_html method to perform special handling of the :rights attribute.  That special handling involves coding an anchor link which opens in a separate tab if the :rights metadata is a url.  Otherwise, the text name of the license is displayed rather than a link.

nomadicoder committed:
Remove "Collection" facet

HYDRASIR-304 #close

rbalekai committed:
Editor Group field for a work should not be a mandatory field

jeremyf commented on app/services/curation_concern/base_actor.rb:
```diff
@@ -4,12 +4,13 @@ module CurationConcern
       @user = user
       @attributes = input_attributes.dup.with_indifferent_access
       @visibility = attributes[:visibility]
+      @cloud_resources= attributes.delete(:cloud_resources.to_s)
```
Why `:cloud_resources.to_s`; Why not just `attributes.delete('cloud_resources')`

jeremyf commented on app/controllers/curation_concern/generic_files_controller.rb:
```diff
@@ -17,6 +18,20 @@ def action_name_for_authorization
   protected :action_name_for_authorization
 
+  def cloud_resources_valid?
+    if params.has_key?(:selected_files) && params[:selected_files].length>1
```
`params[:selected_files].present?` will do the same thing as the above two checks.

banurekha replied:
Need to error out only if more than one file is selected from cloud resource. I dont think present? will do that

jeremyf replied:
Sorry. I misread `>1` as `>0`. You are right.

jeremyf commented on config/browse_everything_providers.yml:
```diff
@@ -0,0 +1,11 @@
+---
```
Are these example provider keys? Because they shouldn't be in the open.

banurekha replied:
I think decision was made to remove this file. I will remove it

jeremyf commented on app/views/curation_concern/images/_form_files_and_links.html.erb:
```diff
@@ -7,6 +7,7 @@
           </legend>
           <%= f.input :files, as: :file, label: 'Upload your file' %>
         </fieldset>
+        <%#render "form_browse_everything", curation_concern: curation_concern, f: f %>
```
This is commented out. Does it need to be here?

banurekha replied:
No. Has to be removed from code. I will delete it on my next pull request.

jeremyf committed:
Adding browse-everything gem dependency

The engine spec/internal regeneration was not including
browse-everything, nor was the engine's gemspec.

HYDRASIR-349

JIRA References:
https://jira.duraspace.org/browse/HYDRASIR-349
https://jira.duraspace.org/browse/HYDRASIR-218

banurekha committed:
Store actual filename as label in fedora since browse-everything gem support file name for google and box

Removing sample browse_everything yml. Removing sky_drive from generator since wont support it for curate

deleting commented code from view file

banurekha committed:
Store actual filename as label in fedora since browse-everything gem support file name for google and box

Removing sample browse_everything yml. Removing sky_drive from generator since wont support it for curate

deleting commented code from view file

nomadicoder commented:
Attempted to upload file with Browse Everything configured for local file system. I selected a file and started upload. Javascript encountered an error: "Uncaught TypeError: Cannot read property 'opts' of null." File was not added.

banurekha commented:
Can you please give me more details? any screenshot.. are you testing on the branch feature/HYDRASIR-281?

nomadicoder commented:
I performed the following in Safari:

```
checkout feature/HYDRASIR-218
bundle install
bundle exec rake regenerate
cd spec/internal
rake jetty:clean
rake jetty:start
bundle exec rails s
```
- Visited `0.0.0.0:3000`
- Created an account, agreed to terms, and updated profile
- Visited `0.0.0.0:3000/remote_files/browse`
- Clicked on `File System`
- Clicked on `features.rb`
- Clicked on "Submit"
- "Submit" button changed to "Loading" and pointer changed to perpetual spinner.

When I tried to submit a file with Chrome, the Javascript console showed the following error:

```
Uncaught TypeError: Cannot read property 'opts' of null behavior.js?body=1:86
  (anonymous function) behavior.js?body=1:86
  fire jquery.js?body=1:3100
  self.fireWith jquery.js?body=1:3212
  done jquery.js?body=1:9311
  callback jquery.js?body=1:9721
```

I also created a local `config/browse_everything_providers.yml` file that would point to an asset directory on my development system. I attempted the same steps with a `.jpg` file and encountered the same problem.

![screen shot 2014-03-27 at 9 11 24 am](https://cloud.githubusercontent.com/assets/14461/2539688/88dd0016-b5ca-11e3-843a-73e3c5c9cfbd.png)

banurekha commented:
browse_everything routes are never meant to be accessed directly. It always sit as modal window on top of form. And hence the error.

So here are the steps to use/test this pull request
- Visited 0.0.0.0:3000
- Created an account, agreed to terms, and updated profile
- Visited 0.0.0.0:3000/concern/generic_works/new (although I keep getting Unregistered predicate: :has_editor: May be side effect of different commit. Not sure)
- select big green button called 'Browse!'
- Select the file from file system or any provider you like
- Click submit
- See a message on number of files selected
- Proceed with filling the form for work
- Agree to terms and submit the form

This should ingested the selected file to fedora.

rbalekai commented:
Unregistered predicate: :has_editor: May be side effect of different commit.

@banurekha, you have to regenerate the spec/internals. I added predicate_mappings.yml file to it. The Group and Editors will expect that to be in place.

nomadicoder commented:
I followed the revised directions and got a server error
![screen shot 2014-03-27 at 9 46 20 am](https://cloud.githubusercontent.com/assets/14461/2540109/8ac56364-b5cf-11e3-8a6d-b7725a1d4c95.png)

banurekha commented:
Found that error later today and pushed the changes to the branch earlier.. Just do a pull.. Should have informed..

banurekha commented:
@rbalekai haven't do that yet.. will let you later once I do that. Thanks for letting me know that

nomadicoder commented:
@banurekha, Thank you. Local files works now. Just have a problem with the browse button where I have to reload the page to make it work. I also need to get my Cloud storage api keys straightened out so I can try the cloud services.

banurekha commented:
This pull request was closed without merging... Should I resend it?

jeremyf commented on CONTRIBUTING.md:
```diff
@@ -152,7 +152,7 @@ However, a well formed Gherkin-syntax story is a suitable proxy; Especially if y
 We encourage the use of [custom rspec matchers](https://www.relishapp.com/rspec/rspec-expectations/v/2-3/docs/custom-matchers/define-matcher) as this helps our specs better convey what is going on.
 
-To find the existing Rspec matchers in this project, you can run `rake spec:show_matchers`.
+To find the existing Rspec matchers in this project, you can run `rake yenta`.
```
I'm not sure how this made it into the commits.

jeremyf commented on app/controllers/catalog_controller.rb:
```diff
@@ -68,9 +78,8 @@ def self.search_config
     # solr fields that will be treated as facets by the blacklight application
     #   The ordering of the field names is the order of the display
-    config.add_facet_field solr_name("human_readable_type", :facetable), label: "Type of Work", limit: 5
+    config.add_facet_field solr_name("human_readable_type", :facetable), label: "Type of Work", :query => human_readable_type_hash, limit: 5
```
Why is this here?
